### DISCUSSION TITLE
How to get hostname is server?

### DISCUSSION THREAD

**Discussion by rmControls** [miguelgrinberg/python-socketio#956] [Q&A] [answered]
I am trying to get hostname and its port where socketio hosted in server code...
I searched in internet and found following code but it is not working...

`socketio.hostname()`

but unfortunately I am getting error `{AttributeError}module 'socketio' has no attribute 'hostname'`

How do we achieve it?

**Comment by miguelgrinberg** [comment]
There is no support for this on this package. You need to look at what options the web server that you use gives you, it else use the litter level networking functions in the Python standard library.

**Comment by rmControls** [comment, marked as answer]
thank you... In `environ` variable I can access `environ['HOST_NAME']`.. which gives you host name...

**Comment by miguelgrinberg** [comment]
This comes from the web server, this package does not set the hostname. You should test it with your production web server and ensure you get reliable information.
